### ISSUE TITLE
Size of dialog windows in the GTK3 version is too small

### ISSUE THREAD

**Issue by marcelpaulo** [BestImageViewer/geeqie#543]
### ISSUE TYPE
* Bug Report

### GEEQIE VERSION
```
Geeqie ac58f0b2
```

### OS / DISTRIBUTION
Xubuntu 17.10

### SUMMARY
The **Edit command results** window size is not being preserved even though preference _Windows -> State -> Remember window positions_ is set.

### STEPS TO REPRODUCE
1. Run any action that causes the **Edit command results** window to appear
1. Resize it
1. Restart geeqie
1. Run any action that causes the **Edit command results** window to appear

The window size should be the one set in **(2)**, but instead it reverts back to the default size.

This doesn't happen with the Ubuntu latest packaged geeqie version, **1.3-1build1**, in other words, with the packaged version, the **Edit command results** window size is preserved.

**EDIT:** Here's my [geeqierc.xml](http://paste.ubuntu.com/25982523/)

**Comment by caclark** [comment]
The `Windows -> State -> Remember window positions ` option only affects the main Geeqie windows. The dialog windows are not influenced by it.
Perhaps the effect you see is one caused by the difference between GTK2 and GTK3. If you compile with
./autogen.sh --disable-gtk3
you will see a difference.

The default sizes of the dialog windows in GTK3 is an annoyance. I think that within a few days I will have completed a patch that gives the option to also preserve the size/position of all the dialog windows.

**Comment by marcelpaulo** [comment]
Indeed, I have just done as you suggested, @caclark, and the sizes of the dialog windows are good. Actually, the look'n'feel of the GTK2 version is overall more polished than the GTK3. Somehow it also felt more responsive.

And I should correct myself: I thought the sizes of the dialog windows were preserved, but the fact is that with GTK2 they have good sizes. I will change the title of the issue to reflect this.

Thanks for clarifying things !

**Comment by caclark** [comment]
Fixed in commit 1caab19584 at geeqie.org

**Comment by marcelpaulo** [comment]
@caclark, I've just built from master with GTK3, but I feel that the size of the dialog windows is still too small. Just to make sure I'm running the latest version that includes your patch: Geeqie d5521e38. To give an example:
![image](https://user-images.githubusercontent.com/10074769/33953448-1f7f902c-e01c-11e7-87c8-a0ba43e5036b.png)
In this particular window, one can't see the names of the files. Another example:
![image](https://user-images.githubusercontent.com/10074769/33953507-50e6b1a4-e01c-11e7-9eb2-6c039ef69179.png)
In this case, one can't see any output from the command.

Wouldn't it be the case to make the GTK3 dialog window sizes the same as the GTK2 ones ? The GTK2 window sizes are quite sensible, one can see enough to know what's happening.

**Comment by marcelpaulo** [comment]
I've rebuilt without GTK3 just to compare the window sizes. Here are the same examples, with GTK2:
![image](https://user-images.githubusercontent.com/10074769/33953678-f3317dcc-e01c-11e7-84ae-2617b2f5a52e.png)
![image](https://user-images.githubusercontent.com/10074769/33953698-04360494-e01d-11e7-9a8e-4b6f37d3bb3b.png)

**Comment by caclark** [comment]
You may resize each of the dialog windows. The title of the window is used to remember its size, so each time you open a dialog of that title it will reappear at the size you set it to last time.

**Comment by marcelpaulo** [comment]
Rebuilt with GTK3, changed the preferences, saved and applied them:
![screenshot_2017-12-13_16-23-32](https://user-images.githubusercontent.com/10074769/33955286-27cafdd8-e022-11e7-9cab-74e39b4e569f.png)
Changed the size of a dialog window:
![screenshot_2017-12-13_16-22-32](https://user-images.githubusercontent.com/10074769/33955315-3bc32aea-e022-11e7-9993-21cd775c9d95.png)
Closed geeqie, did the same operation:
![screenshot_2017-12-13_16-23-11](https://user-images.githubusercontent.com/10074769/33955337-50c9edd4-e022-11e7-86c9-243c4bd5468a.png)
As you can see, @caclark, the window size was not preserved. Just to make sure, here are the preferences again:
![image](https://user-images.githubusercontent.com/10074769/33955385-859281c0-e022-11e7-85ca-3bc3888c95ba.png)

**Comment by marcelpaulo** [comment]
Couldn't understand why, but geeqie crashed after I repeated the operation, so the window size was not preserved. I repeated everything again, this time geeqie didn't crash **AND** the window size was preserved.

I tried a couple of other dialog windows and their size was indeed preserved as you said, @caclark.

I guess I have to investigate now why geeqie crashed when I run my plugin script. But as far as preserving the size of the dialog windows, everything seems OK.

Thanks for shedding light on the matter, and of course, thanks for the patch !
